include/simdjson/portability.h

```diff
@@ -91,16 +79,93 @@ use a 64-bit target such as x64 or 64-bit ARM.")
 //
 // The simdjson users should still have the fallback kernel. It is 
 // slower, but it should run everywhere.
+
+//
+// Enable valid runtime implementations, and select SIMDJSON_BUILTIN_IMPLEMENTATION
+//
+
+//
+// ARM 64-bit (NEON / Other)
+//
+#if SIMDJSON_IS_ARM64
+// Default ARM64 to on: it could be selected at runtime.
+#ifndef SIMDJSON_IMPLEMENTATION_ARM64
+#define SIMDJSON_IMPLEMENTATION_ARM64 1
+#endif
+#if __ARM_NEON
+//
+// NEON
+//
+// We're compiled natively for NEON, so arm64 will *always* work (the program is going
+// to fail in other ways if it's not run on a machine supporting NEON).
+//
+// Set builtin to arm64, and leave fallback defaulted to off.
+// It can still be enabled with SIMDJSON_IMPLEMENTATION_XXX flags, though it's unclear why one
+// would do so).
+//
+#define SIMDJSON_BUILTIN_IMPLEMENTATION arm64
+#endif
+#endif // SIMDJSON_IS_ARM64
+
+//
+// Intel 64-bit (Haswell / Westmere / Other)
+//
 #if SIMDJSON_IS_X86_64
+
+// Default Haswell to on: it could be selected at runtime.
 #ifndef SIMDJSON_IMPLEMENTATION_HASWELL
 #define SIMDJSON_IMPLEMENTATION_HASWELL 1
 #endif
+
+#if __AVX2__ && __BMI__ && __PCLMUL__ && __LZCNT__ && SIMDJSON_IMPLEMENTATION_HASWELL
```

Comment:
I suggest you replace this long list by `#if __AVX2__ && SIMDJSON_IMPLEMENTATION_HASWELL` while documenting that we assume that it implies `__BMI__ && __PCLMUL__ && __LZCNT__`. I believe it is correct in practice. That is, I do not know of a processor that has AVX2, but not the other features. Furthermore, this will resolve the Visual Studio support since I think Visual Studio will only set __AVX2__, not the other macros, when compiling with /arch:AVX2. The Visual Studio documentation clearly states that it enables "other BMI instructions" so the intention is that /arch:AVX2 generates an "haswell or better" binary.